lib/galaxy/tools/actions/__init__.py

```diff
@@ -628,7 +628,7 @@ def set_metadata_defaults( self, output, dataset, tool, on_text, trans, incoming
         """
         if output.actions:
             for action in output.actions.actions:
-                if action.tag == "metadata":
+                if action.tag == "metadata" and action.default:
                     metadata_new_value = fill_template( action.default, context=params ).split(",")
```

Comment:
Should we also check inside `galaxy.util.template.fill_template()` that `template_text` is not None, or is there a legitimate use case?